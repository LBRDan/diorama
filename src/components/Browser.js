import React from 'react';
import PropTypes from 'prop-types';

import styles from './Browser.css';

const Browser = ({ style, className, url }) => (
  <div style={style} className={`${styles.browser} diorama-browser ${className}`}>
    <header className={styles.header}>
      <div className={styles.traffic}>
        <span className={styles.red} />
        <span className={styles.yellow} />
        <span className={styles.green} />
      </div>
      <input className={styles.address} type="text" value={url} disabled />
    </header>
    <iframe src={url} frameBorder="0" title="preview browser" className={styles.iframe} />
  </div>
);

Browser.propTypes = {
  className: PropTypes.string,
  url: PropTypes.string.isRequired,
  style: PropTypes.shape({}),
};

Browser.defaultProps = {
  className: '',
  style: {},
};

export default Browser;
