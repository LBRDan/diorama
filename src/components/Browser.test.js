import React from 'react';
import renderer from 'react-test-renderer';
import Browser from './Browser';

it('renders correctly', () => {
  const tree = renderer.create(<Browser url="https://sambego.be" />).toJSON();
  expect(tree).toMatchSnapshot();
});
