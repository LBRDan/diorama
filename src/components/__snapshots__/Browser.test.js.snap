// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="browser diorama-browser "
  style={Object {}}
>
  <header
    className="header"
  >
    <div
      className="traffic"
    >
      <span
        className="red"
      />
      <span
        className="yellow"
      />
      <span
        className="green"
      />
    </div>
    <input
      className="address"
      disabled={true}
      type="text"
      value="https://sambego.be"
    />
  </header>
  <iframe
    className="iframe"
    frameBorder="0"
    src="https://sambego.be"
    title="preview browser"
  />
</div>
`;
